akka-stream/src/main/scala/akka/stream/impl/CompletedPublishers.scala

```diff
@@ -40,6 +39,36 @@ private[akka] final case class ErrorPublisher(t: Throwable, name: String) extend
   override def toString: String = name
 }
 
+private[akka] final case class SingleElementPublisher[T](value: T, name: String) extends Publisher[T] {
+  import ReactiveStreamsCompliance._
+
+  private[this] class SingleElementSubscription(subscriber: Subscriber[_ >: T]) extends Subscription {
+    private[this] var done: Boolean = false
```

Comment:
Sure, but onSubscribe must `happen-before` the Subscriber can call `request` which must `happen-before` any subsequent such calls or the calls to `onError` or `onComplete` (which would be done in the calling thread (of the subscriber) anyway. So `onSubscribe` must arrange for the safe publication of said `Subscription` to the `Subscriber`.

With code like this it is always possible to be wrong, but I think it holds. :)